subliminal/providers/legendastv.py

```diff
@@ -0,0 +1,466 @@
+            item['type'] = type_map.get(item.get('type'), 'movie')
+            item['imdb_id'] = (lambda m: m.group(1) if m else None)(imdb_re.search(item.get('imdb_id')))
+
+            # Season information might be missing and it should be extracted from 'title_br'
+            if not item.get('season') and item.get('title_br'):
+                item['season'] = (lambda m: m.group(1) if m else None)(season_re.search(item.get('title_br')))
+
+            # Some string fields are actually integers
+            for field in ['season', 'year', 'imdb_id']:
+                item[field] = (lambda v: int(v) if v and v.isdigit() else None)(item.get(field))
+
+            # ignoring episode match since this first step is only about movie/season information
+            if self.matches(params, item, ignore_episode=True):
+                candidates.append(dict(item))
+
+        logger.debug('Titles found: %s', candidates)
+        return candidates
+
+    def query(self, language, video, params):
+        """
+        Returns a list of subtitles based on the input parameters.
+          - 1st step: initial lookup for the movie/show information (see ``search_titles``)
+          - 2nd step: list all candidates all movies/shows from previous step
+          - 3rd step: reject candidates that doesn't match the input parameters (wrong season, wrong episode, etc...)
+          - 4th step: download all subtitles to inspect the 'release name',
+           since each candidate might refer to several subtitles
+
+        :param language: the requested language
+        :param video: the input video
+        :param params: the dictionary with the query parameters
+        :return: a list of subtitles that matches the query parameters
+        :rtype: ``list`` of ``LegendasTvSubtitle``
+        """
+        titles = self.search_titles(params)
+
+        # The language code used by legendas.tv
+        language_code = language.legendastv
+
+        # Regex to extract rating information (number of downloads and rate). e.g.: 12345 downloads, nota 10
+        rating_info_re = re.compile('(\d*) downloads, nota (\d{0,2})')
+
+        # Regex to extract the last update timestamp. e.g.: 25/12/2014 - 19:25
+        timestamp_info_re = re.compile('(\d{1,2}/\d{1,2}/\d{2,4} \- \d{1,2}:\d{1,2})')
+
+        # Regex to identify the 'pack' suffix that candidates might have. e.g.: (p)Breaking.Bad.S05.HDTV.x264
+        pack_name_re = re.compile('^\(p\)')
+
+        # Regex to extract the subtitle_id from the 'href'. e.g.: /download/560014472eb4d/foo/bar
+        subtitle_href_re = re.compile('/download/(\w+)/.+')
+
+        subtitles = []
+        # loop over matched movies/shows
+        for title in titles:
+            # page_url: {server_url}/util/carrega_legendas_busca_filme/{title_id}/{language_code}
+            page_url = '%s/util/carrega_legendas_busca_filme/%s/%d' % (self.server_url, title.get('id'), language_code)
```

Comment:
`title['id']` here, also `title` is a little bit confusing, I'd expect a title to be a string here. Maybe rename titles to candidates?